tests/test_version.py

```diff
@@ -1,19 +1,24 @@
 import os
 import re
+import warnings
 
-from pkg_resources import get_distribution
+try:
+    from importlib.metadata import distribution
+except ImportError:
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
Question: What do these statements do?

Reply:
We run `pytest` with `-W error`, so the pkg_resources deprecation warning turns into a test failure.  This filters out that warning so the tests can still run on python 2.7 and 3.7 where we don't have importlib.metadata.